app/src/test/kotlin/compiler/intermediate_form/DefaultDefaultFunctionDetailsGeneratorTest.kt

```diff
@@ -0,0 +1,348 @@
+        expectedCFGBuilder.addLinksFromAllFinalRoots(
+            CFGLinkType.UNCONDITIONAL,
+            IntermediateFormTreeNode.Add(
+                IntermediateFormTreeNode.RegisterRead(Register.RSP),
+                IntermediateFormTreeNode.Const(16)
+            )
+        )
+
+        val expectedResult = IntermediateFormTreeNode.RegisterRead(Register.RAX)
+        val expected = expectedCFGBuilder.build()
+
+        assert(expected.equalsByValue(result.callGraph))
+        assertEquals(expectedResult, result.result)
+    }
+
+    @Test
+    fun `test gen read direct from memory`() {
+        val memVar: Variable = Variable(Variable.Kind.VALUE, "memVar", Type.Number, null)
+
+        val fdg = DefaultFunctionDetailsGenerator(
+            listOf(),
+            null,
+            IntermediateFormTreeNode.MemoryLabel(""),
+            0u,
+            mapOf(Pair(memVar, VariableLocationType.MEMORY)),
+            IntermediateFormTreeNode.Const(0)
+        )
+
+        var readMemVar = fdg.genRead(memVar, true)
+
+        assertTrue { readMemVar is IntermediateFormTreeNode.MemoryRead }
+
+        readMemVar = readMemVar as IntermediateFormTreeNode.MemoryRead
+        assertTrue { readMemVar.address is IntermediateFormTreeNode.Subtract }
+
+        val left = (readMemVar.address as IntermediateFormTreeNode.Subtract).left
+        val right = (readMemVar.address as IntermediateFormTreeNode.Subtract).right
+
+        assertTrue { left is IntermediateFormTreeNode.RegisterRead }
+        assertEquals(Register.RBP, (left as IntermediateFormTreeNode.RegisterRead).register)
+
+        assertTrue { right is IntermediateFormTreeNode.Const }
+        assertEquals(memoryUnitSize.toLong(), (right as IntermediateFormTreeNode.Const).value)
+    }
+
+    @Test
+    fun `test gen read indirect from memory`() {
+        val memVar: Variable = Variable(Variable.Kind.VALUE, "memVar", Type.Number, null)
+        val displayAddress = IntermediateFormTreeNode.MemoryLabel("display")
+        val depth: ULong = 5u
+        val displayElementAddress = IntermediateFormTreeNode.Add(
+            displayAddress,
+            IntermediateFormTreeNode.Const((memoryUnitSize * depth).toLong())
+        )
+
+        val fdg = DefaultFunctionDetailsGenerator(
+            listOf(),
+            null,
+            IntermediateFormTreeNode.MemoryLabel(""),
+            depth,
+            mapOf(Pair(memVar, VariableLocationType.MEMORY)),
+            displayAddress
+        )
+
+        var readMemVar = fdg.genRead(memVar, false)
+
+        assertTrue { readMemVar is IntermediateFormTreeNode.MemoryRead }
+
+        readMemVar = readMemVar as IntermediateFormTreeNode.MemoryRead
+        assertTrue { readMemVar.address is IntermediateFormTreeNode.Subtract }
+
+        val left = (readMemVar.address as IntermediateFormTreeNode.Subtract).left
+        val right = (readMemVar.address as IntermediateFormTreeNode.Subtract).right
+
+        println(left)
```

Comment:
I noticed this `println` which I believe doesn't belong here

Reply:
Also the file had duplicated `Default` in name, fixed both